decompy/filtercfiles/filter_c_file.py

```diff
@@ -0,0 +1,197 @@
+import os
+import re
+
+
+class FilterC:
+    """
+    Filters C files to our standards. This includes c header files that we find appropraite for machine learning.
+    Filters out the maximum amount of bytes we would like in a file. As of now, this is 7000 bytes.
+    Filters our words that may be too difficult: malloc, FILE, and threading
+    """
+
+    # sets current file directory
+    script_dir = os.path.dirname(os.path.realpath('__file__'))
+
+    # set maximum bytes of 8000, it's around 400 lines of code (including comments and whitespace)
+    MAX_BYTES = 8000
+    MIN_BYTES = 21
+    FILE_TYPE = ".c"
+    # questionable ones: complex, fenv, setjmp, stdalign, stdarg, stdbool, tgmath, uchar
+    C_WHITELIST_HEADERS = ("assert", "complex", "ctype", "errno", "fenv", "float", "inttypes",
+                           "limits", "locale", "math", "signal", "stddef", "stdint", "stdio",
+                           "stdlib", "stdnoreturn", "string", "tgmath", "time", "wchar", "wctype")
+    # bad ones: stdatomic, threads (ruben wants to see threads, malloc, realloc, calloc, and free.
+    C_BLACKLIST = ("file", "malloc", "realloc", "calloc", "free")
+
+    def __init__(self):
+        """
+        init
+
+        Initializes an object to filter C files.
+        """
+
+    @staticmethod
+    def check_byte_size(file, preferred_max_size=MAX_BYTES, preferred_min_size=MIN_BYTES):
+        """
+        Finds the file size and tests it against the preferred_size in bytes. The default is 7000 bytes.
+
+        :param file: the file path to test against
+        :type: str
+        :param preferred_max_size: the preferred size to search for, defaults to 7000.
+        :type: int
+        :param preferred_min_size: the preferred minimum size to search for, defaults to 21 (for `int main(){return 0;}`)
+        :type: int
+        :return: boolean
+        """
+
+        # combine to get the correct file
+        # file = os.path.join(FilterC.script_dir, file)
+        file_size = os.stat(file).st_size
+
+        return preferred_max_size > file_size >= preferred_min_size
+
+    @staticmethod
+    def check_blacklisted_words(line, blacklisted=C_BLACKLIST):
+        """
+        Lowercases the line to evaluate, and returns false if any blacklisted word is found.
+        :param: the string from a file.
+        :type: str
+        :param: the blacklisted array.
+        :type: str arr
+        :return: boolean
+        """
+
+        line = line.lower()
+        return any(word in line for word in blacklisted)
```

Comment:
For efficiency, we should consider using a Trie here later on. It would make the comparisons much faster, especially with long blacklists. We can do this later though